### PR TITLE
Fix panics from self-referential amf3 objects

### PR DESCRIPTION
**Pull request by CUB3D**
This fixes some instances of the "failed to get object" panic in ruffle by no longer trying to handle value cycles 

Now all object values are tagged with an ObjectId and the new Value::Amf3ObjectReference's refer back to this so consumers can choose how to handle cycles (e.g. in Ruffles case, via the GC)

Also moves raw-amf/byteobject tests into their own dir as they aren't part of the minerva tests and changed them from .sol -> .amf because they aren't lso files. Adds a test for amf3 cycles and improves the lso-to-json tool to support bulk-test json recreation and generating raw-amf tests

This fixes the following (requires a Ruffle patch as well):
https://github.com/ruffle-rs/ruffle/issues/17902 - loads
https://github.com/ruffle-rs/ruffle/issues/17782 - doesn't panic on load, avm2 error due to null object
https://github.com/ruffle-rs/ruffle/issues/17709 - loads
https://github.com/ruffle-rs/ruffle/issues/17336 - no swf to test, but probably a dupe of https://github.com/ruffle-rs/ruffle/issues/17782
https://github.com/ruffle-rs/ruffle/issues/14329 - doesn't panic on load, avm2 error due to null object
https://github.com/ruffle-rs/ruffle/issues/13748 - loads
https://github.com/ruffle-rs/ruffle/issues/10174 - doesn't panic on load, avm2 error due to null object
https://github.com/ruffle-rs/ruffle/issues/10520 - loads

### CODE REVIEW

File: flash-lso/src/amf3/read.rs [hunk 1]
```diff
@@ -151,12 +151,17 @@ fn parse_element_int(i: &[u8]) -> AMFResult<'_, Rc<Value>> {
...
     /// The table used to cache repeated trait definitions
     pub trait_reference_table: Vec<ClassDefinition>,
+
     /// The table used to cache repeated objects
     pub object_reference_table: Vec<Rc<Value>>,
+
     /// Encoders used for handling externalized types
     pub external_decoders: HashMap<String, ExternalDecoderFn>,
+
+    object_id: i64,
```
Comment (adrian17) [inline flash-lso/src/amf3/read.rs:164]: can you please document this field for the future? (including why it's not `ObjectId`)
Also, any reason why it differs in signed-ness from ObjectId?

Comment (CUB3D) [reply flash-lso/src/amf3/read.rs:166]: Added docs. 
It doesn't differ from ObjectId, they are both `i64`s?

Comment (adrian17) [reply flash-lso/src/amf3/read.rs:166]: My bad, could have sworn I saw `u64` somewhere :/

**Comment by torokati44** [comment]
Wowza, 87 changed files?! Something must have gotten haywire... Though it seems, it's mostly "line terminator at end of file" stuff...? Still, I don't think those belong in this PR, even if intentional.

**Comment by CUB3D** [comment]
> Wowza, 87 changed files?! Something must have gone haywire... Though it seems, it's mostly "line terminator at end of file" stuff...? Still, I don't think those belong in this PR, even if intentional.

So *some* of those are intentional, as the enum format has changed so the files needed regenerating. But yes I was missing the newlines on the unchanged ones, fixed now.

**Comment by torokati44** [comment]
Thank you! :tada:

**Pull request merged**